crates/router/src/connector/stripe/transformers.rs

```diff
@@ -3432,6 +3432,42 @@ impl TryFrom<&types::PaymentsCancelRouterData> for CancelRequest {
     }
 }
 
+#[derive(Debug, Serialize)]
+pub struct UpdateMetadataRequest {
+    #[serde(flatten)]
+    pub metadata: StripeUpdateMetadata,
+}
+
+#[derive(Debug, Serialize)]
+pub struct StripeUpdateMetadata {
+    #[serde(rename = "metadata[order_id]")]
+    pub order_id: Option<String>,
+}
+
+impl TryFrom<&types::PaymentsUpdateMetadataRouterData> for UpdateMetadataRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsUpdateMetadataRouterData) -> Result<Self, Self::Error> {
+        let metadata = item
+            .request
+            .metadata
+            .as_ref()
+            .get_required_value("request.metadata")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "request.metadata",
+            })?
+            .clone()
+            .expose();
+
+        let order_id = metadata
+            .get("order_id")
+            .and_then(|val| val.as_str())
+            .map(String::from);
+        Ok(Self {
+            metadata: StripeUpdateMetadata { order_id },
```

Comment:
Just curious, would we still end up sending the request if the `order_id` field is null / absent?